src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -13,4 +13,6 @@ class CompilationUnit(val source: SourceFile) {
   var tpdTree: tpd.Tree = tpd.EmptyTree
 
   def isJava = source.file.name.endsWith(".java")
+  
+  var pickled: Array[Byte] = Array()
```

Comment:
As some phases could want to run after `pickler` and add additional information:
 - Maybe ArrayBuffer? Though I'd prefer to not allow to mutate already pickled info, and export TastyPickler itself. This will allow to add new sections easily.
 - and an `addrOfTree: Tree => Option[Addr]` for them to correlate with already pickled info

This will also allow to make PositionPickler a separate phase.
@odersky If you agree, I'll do this in a separate PR